create.ps1

```diff
@@ -1,200 +1,220 @@
-    externalId     = $p.ExternalId
-    geslacht       = $gender
-}
+try {
+    # Create account object from mapped data and set the correct account reference
+    $account = $actionContext.Data;
+    $person = $personContext.Person;
 
-#correlation
-$correlationField = 'externalId'
-$correlationValue = $p.ExternalID
+    <#
+    # Make sure module is imported
+    $moduleName = "PSSQLite"
 
-Try {     
-     
-    $query = "SELECT $correlationField,achternaam FROM persons WHERE $correlationField = '$correlationValue'"
-    $correlationCheckResult = Invoke-SqliteQuery -Query $query -DataSource $database -Verbose:$verbose
+    # If module is imported say that and do nothing
+    if (Get-Module -Verbose:$false | Where-Object { $_.Name -eq $ModuleName }) {
+        Write-Verbose "Module [$ModuleName] is already imported."
+    }
+    else {
+        # If module is not imported, but available on disk then import
+        if (Get-Module -ListAvailable -Verbose:$false | Where-Object { $_.Name -eq $ModuleName }) {
+            $module = Import-Module $ModuleName -Verbose:$false
+            Write-Verbose "Imported module [$ModuleName]"
+        }
+        else {
+            # If the module is not imported, not available and not in the online gallery then abort
+            throw "Module [$ModuleName] is not available. Please install the module using: Install-Module -Name [$ModuleName] -Force"
+        }
+    }
+    #>
 
+    # Check if we should try to correlate the account
+    if ($actionContext.CorrelationConfiguration.Enabled) {
+        $correlationField = $actionContext.CorrelationConfiguration.accountField
+        $correlationValue = $actionContext.CorrelationConfiguration.accountFieldValue
 
-    #Check correlation before create
-    if ($correlationCheckResult.externalId -eq $correlationValue) {
-        $aRef = ($correlationCheckResult.$correlationField)
+        if ($null -eq $correlationField) {
+            Write-Warning "Correlation is enabled but not configured correctly."
+        }
 
-        $query = "UPDATE persons 
-                    SET email = @email
-                    ,achternaam = @achternaam
-                    ,voornaam = @voornaam
-                    ,tussenvoegsel = @tussenvoegsel
-                    ,gebruikersnaam = @gebruikersnaam
-                    ,geslacht = @geslacht
-                    ,createtime = datetime()
-                    WHERE externalId = @externalId"
+        # Write logic here that checks if the account can be correlated in the target system
+        # Requesting authorization token
 
-        $sqlParameters = $account.psobject.properties | ForEach-Object -begin { $h = @{} } -process { $h."$($_.Name)" = $_.Value } -end { $h }
+        try {
+            $query = "SELECT $correlationField,achternaam,gebruikersnaam FROM persons WHERE $correlationField = '$correlationValue'"
+            $correlationCheckResult = Invoke-SqliteQuery -Query $query -DataSource $database -Verbose:$verbose
```

Comment:
Its possible to have multiple rows with the same externalId within the database. So maybe we should add another check to deal with that situation?